Update imports to only Packages or Modules

pferate opened the pull request:
Also cleaned up some nested attribute access.

Resolves issue #556

pferate committed:
Update imports to only Packages or Modules

Also cleaned up some nested attribute access.

pferate commented:
I got all but one line down to under 80 characters.  Without assigning temporary variables to shorten the line, it will take some refactoring beyond the scope of this PR to clean it up.

I've posted some options in #556 on how to proceed.

dhermes commented:
Just assign a temporary variable. Anything else isn't worth the headache

pferate commented:
Will do.

I want to add a TODO to fix that, but need to reference something.  I'm thinking of adding an issue to refactor the built-in credentials.  It started as a small refactor in my head, but it's becoming bigger as I think about it more.

pferate commented:
Rebased branch, added issue #564, and updated this PR.

jonparrott commented:
@pferate this branch has conflicts, can you update?

pferate commented:
@jonparrott: Updated

theacodes commented on oauth2client/contrib/django_util/__init__.py:
```diff
@@ -84,9 +84,9 @@
+   from oauth2client.contrib.django_util import site
 
-   urlpatterns += [url(r'^oauth2/', include(oauth2_urls))]
+   urlpatterns += [url(r'^oauth2/', include(site.oauth2_urls))]
```
Since this in an example, it probably makes sense to leave this as it was. @waprin

theacodes commented on oauth2client/contrib/django_util/__init__.py:
```diff
@@ -97,9 +97,9 @@
    :name: views_required
 
-   from oauth2client.contrib.django_util.decorators import oauth_required
+   from oauth2client.contrib.django_util import decorators
```
same here.

nathanielmanistaatgoogle commented on oauth2client/contrib/appengine.py:
```diff
@@ -56,6 +49,8 @@
 
 logger = logging.getLogger(__name__)
 
+Credentials = client.Credentials
```
Why this?

pferate replied:
Left overs, from when there were indirect references.  There were some references in the code to `contrib.appengine.Credentials` which was just `client.Credentials`.

Removing.

nathanielmanistaatgoogle commented on oauth2client/contrib/dictionary_storage.py:
```diff
@@ -14,11 +14,13 @@
 
 
-class DictionaryStorage(Storage):
+OAuth2Credentials = client.OAuth2Credentials
```
Why this?

pferate replied:
Same case as the one in `appengine.py`.  Removing.

nathanielmanistaatgoogle commented on tests/contrib/test_appengine.py:
```diff
@@ -520,42 +505,42 @@ def test_delete_ndb(self):
+            appengine.CredentialsModel, 'foo', 'credentials')
 
         # First DB, then NDB
         self.assertEqual(None, storage.get())
```
For a future cleanup: `assertIsNone`.

pferate replied:
Already planned for phase 2 of the Pytest migration.

nathanielmanistaatgoogle commented on tests/test_jwt.py:
```diff
@@ -328,5 +325,5 @@ def setUp(self):
     def test_true(self):
-        self.assertEqual(True, HAS_OPENSSL)
-        self.assertEqual(True, HAS_CRYPTO)
+        self.assertEqual(True, client.HAS_OPENSSL)
```
Also possibly in a future cleanup: `assertTrue` or `assertIs` (depending on the semantics of the system under test).

pferate replied:
Planned for phase 2.  Will be `assert client.HAS_OPENSSL is True`

nathanielmanistaatgoogle commented on tests/test_service_account.py:
```diff
@@ -103,7 +102,9 @@ def _from_json_keyfile_name_helper(payload, scopes=None,
             with open(filename, 'w') as file_obj:
                 json.dump(payload, file_obj)
-            return ServiceAccountCredentials.from_json_keyfile_name(
+            # TODO(issue 564): Cleanup Credentials access
```
One way to do this without introducing a local field is to just wrap it in parentheses:

```
return (
    service_account.ServiceAccountCredentials
    .from_json_keyfile_name)(
    <arguments>)
```

. Up to you though.

pferate replied:
For some reason, that style of wrapping escaped me; I was trying that with parenthesis before and after the dot, but not without them.  Updating per your suggestion.

nathanielmanistaatgoogle commented:
Oof, that's a long code review. I keep forgetting that we have this much code in the library.

Looks mostly good; just those few observations.

pferate commented:
Thanks @nathanielmanistaatgoogle!

I've updated the PR, per your suggestions and rebased again.

I mentioned this in another PR, but the `flake8` test is failing due to a bug in `flake8-import-order` (0.9 released yesterday).  I filed an issue, so hopefully it will be fixed shortly.

nathanielmanistaatgoogle commented:
Link to the `flake8` bug you filed?

pferate commented:
https://github.com/PyCQA/flake8-import-order/issues/80

pferate commented:
The `flake8` bug has been resolved and released in 0.9.1.
